packages/salesforcedx-vscode-core/test/vscode-integration/commands/templates/internalCommandUtils.test.ts

```diff
@@ -0,0 +1,106 @@
+ */
+
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { expect } from 'chai';
+import * as fs from 'fs';
+import * as path from 'path';
+import { SinonStub, stub } from 'sinon';
+import { Uri } from 'vscode';
+import {
+  FileInternalPathGatherer,
+  InternalDevWorkspaceChecker
+} from '../../../../src/commands/templates/internalCommandUtils';
+import { SfdxCoreSettings } from '../../../../src/settings/sfdxCoreSettings';
+
+describe('Internal Command Utilities', () => {
+  describe('Internal Workspace Checker', () => {
+    let settings: SinonStub;
+
+    beforeEach(() => {
+      settings = stub(SfdxCoreSettings.prototype, 'getInternalDev');
+    });
+
+    afterEach(() => {
+      settings.restore();
+    });
+
+    it('Should always return false', async () => {
+      settings.returns(false);
+      const internalDevWSChecker = new InternalDevWorkspaceChecker();
+      expect(internalDevWSChecker.check()).to.be.eql(false);
+    });
+
+    it('Should always return true', async () => {
+      settings.returns(true);
+      const internalDevWSChecker = new InternalDevWorkspaceChecker();
+      expect(internalDevWSChecker.check()).to.be.eql(true);
+    });
+  });
+
+  describe('File Internal Path Gatherer', () => {
+    let existsSyncStub: sinon.SinonStub;
+    let lstatSyncStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      existsSyncStub = stub(fs, 'existsSync');
+      lstatSyncStub = stub(fs, 'lstatSync');
+    });
+
+    afterEach(() => {
+      existsSyncStub.restore();
+      lstatSyncStub.restore();
+    });
+
+    it('Should return Continue', async () => {
+      existsSyncStub.returns(true);
+      const testDir = path.join('path', 'to', 'outside', 'dir');
+      lstatSyncStub.returns({
+        isDirectory() {
+          return true;
+        }
+      });
+
+      const folderPathGatherer = new FileInternalPathGatherer(
+        Uri.parse(testDir)
+      );
+      const response = (await folderPathGatherer.gather()) as ContinueResponse<{
+        outputdir: string;
+      }>;
+      expect(response.type).to.equal('CONTINUE');
+      expect(response.data.outputdir).contains(testDir);
+    });
+
+    it('Should return Cancel if path is not a directory', async () => {
+      existsSyncStub.returns(true);
+      const testDir = Uri.parse('file:///path/to/outside/dir');
+      lstatSyncStub.returns({
+        isDirectory() {
+          return false;
+        }
+      });
+
+      const folderPathGatherer = new FileInternalPathGatherer(testDir);
+      const response = await folderPathGatherer.gather();
+      expect(response.type).to.equal('CANCEL');
+    });
+
+    it('Should return Cancel if path does not exist', async () => {
+      existsSyncStub.returns(false);
+      const testDir = Uri.parse('file:///path/to/outside/dir');
+      lstatSyncStub.returns({
+        isDirectory() {
+          return false;
```

Comment:
Very minor nit, but I might make this return true so we guarantee we're not returning cancel because it's not a directory.

Reply:
The reason I set it to false is because this scenario tests for a path that doesn't exist. If the path does not exist (line 93) then isDirectory needs to return false.
I think the other tests cover the rest of the scenarios where the path exists and it's a directory or is a file. Let me know if there's other scenarios you're thinking about.